client-v2/src/app/pages/settings/account/account.component.ts

```diff
@@ -38,23 +46,56 @@ export class SettingsAccountComponent implements OnDestroy {
         this.toast.info('Profile photos are not yet implemented.')
     }
     onDeleteAccount() {
-        const message =
-            'Do you really want to delete your account? Everything will be deleted. This action is irreversable!'
-        if (!confirm(message)) return
+        // @TODO: implement this with cdk stepper
+        const dialogRef = this.dialogService.confirm({
+            title: 'Delete account',
+            text: `Are you sure you want to delete your account? All data associated with your account will be lost. This action cannot be undone.`,
+            buttons: [
+                { text: 'Cancel' },
+                {
+                    text: 'I understand, delete',
+                    resCode: 'Delete',
+                    className: 'button--danger',
+                },
+            ],
+        })
+
+        dialogRef.closed.subscribe(res => {
+            if (res != 'Delete') return
 
-        const password = prompt('Type in your password.')
-        if (!password) return
+            moveToMacroQueue(() => {
+                const password = prompt('Type in your password.')
+                if (!password) return
 
-        this.store.dispatch(accountActions.deleteAccount({ password }))
+                this.store.dispatch(accountActions.deleteAccount({ password }))
+            })
+        })
     }
     onLogout() {
-        if (confirm('Do you want to logout?')) this.store.dispatch(authActions.logout())
+        this.store.dispatch(authActions.logout())
     }
 
-    activeForm: null | 'email' | 'password' = null
-    setFormActive(form: typeof this.activeForm) {
-        this.activeForm = form
+    @ViewChild('emailForm') emailForm!: TemplateRef<HTMLElement>
+    @ViewChild('passwordForm') passwordForm!: TemplateRef<HTMLElement>
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    modalRef: null | DialogRef<any, any> = null
+    openFormModal(form: null | 'email' | 'password') {
+        if (!form) return
+
+        const formRefMap = {
+            email: this.emailForm,
+            password: this.passwordForm,
+        }
+
+        this.modalRef = this.dialogService.open(formRefMap[form], {})
+        this.modalRef.closed.subscribe(() => this.openFormModal(null))
```

Comment:
Lets just reset the `modalRef` to `null` instead of calling the method with `null` which doesn't do anything anyway.